src/daemon/config.js

```diff
@@ -238,19 +329,19 @@ async function checkPorts (ipfsd) {
     return checkPortsArray(ipfsd, [].concat(config.Addresses.API, config.Addresses.Gateway))
   }
 
-  const configApiMa = parseCfgMultiaddr(config.Addresses.API)
-  const configGatewayMa = parseCfgMultiaddr(config.Addresses.Gateway)
+  const configApiMa = parseMultiaddr(config.Addresses.API)
+  const configGatewayMa = parseMultiaddr(config.Addresses.Gateway)
 
   const isApiMaDaemon = await checkIfAddrIsDaemon(configApiMa.nodeAddress())
   const isGatewayMaDaemon = await checkIfAddrIsDaemon(configGatewayMa.nodeAddress())
 
   if (isApiMaDaemon && isGatewayMaDaemon) {
     logger.info('[daemon] ports busy by a daemon')
-    return
+    return true
```

Comment:
> Returns true if ports are cleared for IPFS to start.

Do we want to return true if the ports are busy?

Reply:
Ah, this is a specific case: if ports are busy by a **daemon**, it shouldn't be a problem since we connect to it anyways (via the API). The problem is when the ports are busy by something else.